core/analytics/src/main/kotlin/com/wire/android/feature/analytics/model/AnalyticsEvent.kt

```diff
@@ -129,6 +146,45 @@ interface AnalyticsEvent {
         }
     }
 
+    data class RecentlyEndedCallEvent(val metadata: RecentlyEndedCallMetadata) : AnalyticsEvent {
+        override val key: String = CALLING_ENDED
+
+        override fun toSegmentation(): Map<String, Any> {
+            return mapOf(
+                CALLING_ENDED_IS_TEAM_MEMBER to metadata.isTeamMember,
+                CALLING_ENDED_CALL_SCREEN_SHARE to metadata.callDetails.screenShareDurationInSeconds,
+                CALLING_ENDED_UNIQUE_SCREEN_SHARE to metadata.callDetails.callScreenShareUniques,
+                CALLING_ENDED_CALL_DIRECTION to metadata.toCallDirection(),
+                CALLING_ENDED_CALL_DURATION to metadata.callDetails.callDurationInSeconds,
+                CALLING_ENDED_CONVERSATION_TYPE to metadata.toConversationType(),
+                CALLING_ENDED_CONVERSATION_SIZE to metadata.conversationDetails.conversationSize,
+                CALLING_ENDED_CONVERSATION_GUESTS to metadata.conversationDetails.conversationGuests,
+                CALLING_ENDED_CONVERSATION_GUESTS_PRO to metadata.conversationDetails.conversationGuestsPro,
+                CALLING_ENDED_CALL_PARTICIPANTS to metadata.callDetails.callParticipantsCount,
+                CALLING_ENDED_END_REASON to metadata.callEndReason,
+                CALLING_ENDED_CONVERSATION_SERVICES to metadata.callDetails.conversationServices,
+                CALLING_ENDED_AV_SWITCH_TOGGLE to metadata.callDetails.callAVSwitchToggle,
+                CALLING_ENDED_CALL_VIDEO to metadata.callDetails.callVideoEnabled,
+            )
+        }
+
+        private fun RecentlyEndedCallMetadata.toCallDirection(): String {
+            return if (callDetails.isOutgoingCall) {
+                "outgoing"
+            } else {
+                "incoming"
+            }
+        }
+
+        private fun RecentlyEndedCallMetadata.toConversationType(): String {
+            return when (conversationDetails.conversationType) {
+                Conversation.Type.ONE_ON_ONE -> "one_to_one"
+                Conversation.Type.GROUP -> "group"
+                else -> "unknown"
```

Comment:
Is "unknown" specified anywhere?

Reply:
This `unknown` would happen only if the conversation type would be `OWN` or `CONNECTION_PENDING` to which we cant call, so it will never happen in fact, I will change it up to an error instead